docs/reference-guides/block-api/block-bindings.md

```diff
@@ -0,0 +1,285 @@
+# Bindings
+
+The Block Bindings API lets you “bind” dynamic data to the block’s attributes, which are then reflected in the final HTML markup that is output to the browser on the front end.
+
+An example could be connecting an Image block `url` attribute to a function that returns random images from an external API.
+
+```html
+<!-- wp:image {
+	"metadata":{
+		"bindings":{
+			"url":{
+				"source":"my-plugin/get-random-images"
+			}
+		}
+	}
+} -->
+```
+
+
+## Compatible blocks and their attributes
+
+Right now, not all block attributes are compatible with block bindings. This is some ongoing effort to increase this compatibility, but for now, this is the list:
+
+| Supported Blocks    | Supported Attributes |
+| -------- | ------- |
+| Paragraph  | content    |
+| Heading | content     |
+| Image    | ID, url, title, alt    |
+| Button    | text, url, linkTarget, rel    |
+
+## Registering a custom source
+
+Registering a source requires defining at least `name` and a `callback` function that gets a value from the source and passes it back to a block attribute.
+
+Once a source is registered, any supporting block's `metadata.bindings` attribute can be configured to read a value from that source.
+
+Registration can be done on the server via PHP or in the editor via JavaScript, and both can coexist.
+
+The label defined in server registration will be overridden by the label defined in the editor.
+
+### Server registration
+
+Server registration allows applying a callback that will be executed on the frontend for the defined bound attribute.
+
+The function to register a custom source is `register_block_bindings_source($name, $args)`.
+
+- `name`: `string` that sets the unique ID for the custom source.
+- `args`: `array` that contains:
+    - `label`: `string` with the human-readable name of the custom source.
+    - `uses_context`: `array` with the block context that is passed to the callback (optional).
+    - `get_value_callback`: `function` that will run on the bound block render function. It accepts three arguments: `source_args`, `block_instance` and `attribute_name`. This value can be overriden with the filter `block_bindings_source_value`.
+
+Note that `register_block_bindings_source()` should be called from a handler attached to the `init` hook.
+
+Here is an example:
+
+```php
+add_action(
+	'init',
+	function() {
+		register_block_bindings_source(
+			'wpmovies/visualization-date',
+			array(
+				'label'              => __( 'Visualization Date', 'custom-bindings' ),
+				'get_value_callback' => function( array $source_args, $block_instance ) {
+					$post_id = $block_instance->context['postId'];
+					$visualization_date = get_post_meta( $post_id, 'wp_movies_visualization_date', true );
+					if ( ! $visualization_date ) {
+						$visualization_date = date("m/d/Y");
```

Comment:
I recommend running Prettier on this file to ensure WordPress coding styles are followed.

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/66251/commits/a2e5a7ac2d1ff2eafbb1688ebf820bc9b20bc268